core/MyGatewayTransportEthernet.cpp

```diff
@@ -150,19 +150,10 @@ void gatewayTransportRenewIP(void)
 	 3 - rebind failed
 	 4 - rebind success
 	 */
-	static uint32_t _nextIPRenewal = hwMillis() + MY_IP_RENEWAL_INTERVAL_MS;
-	const uint32_t now = hwMillis();
-
-	// http://playground.arduino.cc/Code/TimingRollover
-	if ((int32_t)(now - _nextIPRenewal) < 0) {
-		return;
-	}
 	if (Ethernet.maintain() & ~(0x06)) {
 		GATEWAY_DEBUG(PSTR("!GWT:TRC:IP RENEW FAIL\n"));
-		return;
 	}
 	_w5100_spi_en(false);
```

Comment:
Not sure why `_w5100_spi_en(false)` was only called in case of successful renewal before. IF it should be called at all, my guess is that it should be called regardless whether renewal was successful or failed

Reply:
gatewayTransportRenewIP is only called from gatewayTransportAvailable.
gatewayTransportAvailable starts with _w5100_spi_en(true);
With your change we return to gatewayTransportAvailable with _w5100_spi_en set to false.
This is likely to break the remaining code in gatewayTransportAvailable.

I would remove any call to _w5100_spi_en() from gatewayTransportRenewIP, so it never messes with the state of the SPI_EN line.

Reply:
To be clear: without my change, we return to gatewayTransportAvailable with SPI_EN set to false when the renewal was successful (which it it most of the time).